client/asset/interface.go

```diff
@@ -72,7 +72,7 @@ type Wallet interface {
 	// Fund selects coins for use in an order. The coins will be locked, and will
 	// not be returned in subsequent calls to Fund or calculated in calls to
 	// Available, unless they are unlocked with ReturnCoins.
-	FundOrder(*Order) (Coins, error)
+	FundOrder(*Order) (Coins, []dex.Bytes, error)
```

Comment:
Please doc the new output.
Should probably also specify if coins with no redeem script should append a `nil` or an empty `dex.Bytes`.  It affects the marshaling of `msgjson.Coin`.
Presently dcr's `fund` method's use of `hex.DecodeString` ends up putting an empty non-nil slice, which seems to be fine.  Let's just be sure that's desirable and specify it in the interface.

Reply:
Looks like either nil or empty dex.Bytes work but my main concern is callers having to validate that len(coins) == len(redeemScripts). I'm considering returning a `map[Coin]dex.Bytes` to prevent implementations from erroneously returning unequal number of coins and redeem scripts.

Reply:
We chatted on matrix, and settled on a `[]struct{Coin, dex.Bytes}`.  My feeling was that the map was confusing and the interface key isn't ideal in my opinion.